File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
         ListNode dummy = new ListNode(0);
        dummy.next = head;
        int count=0;
        ListNode curr=head;
        while (curr!=null){
            count ++;
            curr= curr.next;
        }
        count-=n;
        curr=dummy;
        for (int i=0; i<count; i++){
            curr=curr.next;
        }
        curr.next=curr.next.next;
        return dummy.next;


    }
}

File: Data Structures & Algorithms/maximum-twin-sum-of-a-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        ListNode* fast= head, *slow= head;
        while(fast->next && fast){
            slow= slow->next;
            fast= fast->next->next;
        }
        ListNode *prev= nullptr, * curr= slow;
        while(curr){
            ListNode* next= curr->next;
            curr->next= prev;
            prev= curr;
            curr= next;
        }
        int ans=0;
        ListNode* first=head, *second=prev;
        while(second){
            ans= max(ans, first->val+second->val);
            first= first->next;
            second= second->next;
        }
        return ans;
    }
};